mmpose/models/keypoint_heads/heatmap_3d_head.py

```diff
@@ -0,0 +1,222 @@
+import numpy as np
+import torch.nn as nn
+
+from mmpose.core.post_processing import flip_back, transform_preds
+from ..registry import HEADS
+from .top_down_simple_head import TopDownSimpleHead
+
+
+def _get_max_preds_3d(heatmaps):
+    """Get keypoint predictions from 3D score maps.
+
+    Note:
+        batch_size: N
+        num_keypoints: K
+        heatmap depth size: D
+        heatmap height: H
+        heatmap width: W
+
+    Args:
+        heatmaps (np.ndarray[N, K, D, H, W]): model predicted heatmaps.
+
+    Returns:
+        tuple: A tuple containing aggregated results.
+        - preds (np.ndarray[N, K, 3]): Predicted keypoint location.
+        - maxvals (np.ndarray[N, K, 1]): Scores (confidence) of the keypoints.
+    """
+    assert isinstance(heatmaps, np.ndarray), \
+        ('heatmaps should be numpy.ndarray')
+    assert heatmaps.ndim == 5, 'heatmaps should be 5-ndim'
+
+    N, K, D, H, W = heatmaps.shape
+    heatmaps_reshaped = heatmaps.reshape((N, K, -1))
+    idx = np.argmax(heatmaps_reshaped, 2).reshape((N, K, 1))
+    maxvals = np.amax(heatmaps_reshaped, 2).reshape((N, K, 1))
+
+    preds = np.tile(idx, (1, 1, 3)).astype(np.float32)
```

Comment:
In L38-40, values in preds all are overriden. So,
```python
preds = np.zeros((N, K, 3), dtype=np.float32)
```

Reply:
OK, fixed